lib/alephant/broker/response/base.rb

```diff
@@ -48,10 +52,19 @@ def add_no_cache_headers
           log
         end
 
+        def add_etag_allow_header
+          headers.merge!("Access-Control-Expose-Headers" => "ETag")
+        end
+
         def component_not_modified(headers, request_env)
```

Comment:
How do you mean class level method?

Isn't the headers in this, driven by the instance? As it is component headers, then it will be on the state of those headers, so they will have different ETags or Last Modified dates.

I did look at using the `headers` property of the class, so we don't pass it in, but it looks like the component headers get merged in after this function is called. In the singular asset, this is called before merging component headers in. The whole logic for singular and batch initialisation, and then calling the setup method, during, is all a bit messy.